notebooks/NonClassicalSearch.ipynb

```diff
@@ -0,0 +1,1027 @@
+       "&emsp;_current_ &larr; MAKE\\-NODE(_problem_.INITIAL\\-STATE)  \n",
+       "&emsp;__loop do__  \n",
+       "&emsp;&emsp;&emsp;_neighbor_ &larr; a highest\\-valued successor of _current_  \n",
+       "&emsp;&emsp;&emsp;__if__ _neighbor_.VALUE &le; _current_.VALUE __then return__ _current_.STATE  \n",
+       "&emsp;&emsp;&emsp;_current_ &larr; _neighbor_\n",
+       "\n",
+       "---\n",
+       "__Figure ??__ The hill\\-climbing search algorithm, which is the most basic local search technique. At each step the current node is replaced by the best neighbor; in this version, that means the neighbor with the highest VALUE, but if a heuristic cost estimate _h_ is used, we would find the neighbor with the lowest _h_."
+      ],
+      "text/plain": [
+       "<IPython.core.display.Markdown object>"
+      ]
+     },
+     "execution_count": 17,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%%python\n",
+    "from notebookUtils import *\n",
+    "pseudocode('Hill Climbing')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The implementation of the above pseudo code can be viewed [here](https://github.com/aimacode/aima-java/blob/AIMA3e/aima-core/src/main/java/aima/core/search/local/HillClimbingSearch.java). \n",
+    "\n",
+    "This algorithm halts if it reaches a plateau where the best successor has the same value as the current state. It might be a good idea to allow sideways move in the hope that plateau is really a shoulder, but we must take care. An infinite loop can occur whenever the algorithm reaches a flat local maximum that is not a shoulder. One common solution is to put a limit on the number of consecutive sideways moves allowed.\n",
+    "\n",
+    "Let's try to solve 8-Queens problem using Hill-climbing search."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 62,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Action[name=moveQueenTo, location=(2, 1)]\n",
+      "Action[name=moveQueenTo, location=(1, 3)]\n",
+      "Action[name=moveQueenTo, location=(4, 1)]\n",
+      "Action[name=moveQueenTo, location=(6, 2)]\n",
+      "Action[name=moveQueenTo, location=(2, 0)]\n",
+      "Search Outcome=FAILURE\n",
+      "Final State=\n",
+      "--Q-----\n",
+      "----Q---\n",
+      "------Q-\n",
+      "-Q------\n",
+      "---Q----\n",
+      "-----Q--\n",
+      "Q-------\n",
+      "-------Q\n",
+      "\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "null"
+      ]
+     },
+     "execution_count": 62,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import aima.core.agent.Action;\n",
+    "import aima.core.environment.nqueens.NQueensBoard;\n",
+    "import aima.core.environment.nqueens.NQueensFunctions;\n",
+    "import aima.core.environment.nqueens.QueenAction;\n",
```

Comment:
Try reducing the number of import statements. For eg: The above three statements can be reduced to:
````java
import aima.core.environment.nqueens.*;
````
This makes the code look cleaner and a lot less intimidating.